python/lsst/daf/butler/datastores/chainedDatastore.py

```diff
@@ -230,7 +248,7 @@ def exists(self, ref):
                 return True
         return False
 
-    def get(self, ref, parameters=None):
+    def get(self, ref: DatasetRef, parameters: Mapping[str, Any] = None) -> Any:
```

Comment:
Looks like this needs to be `Optional[Mapping[str, Any]]`.  Did mypy maybe not get enabled on this file?

Reply:
It was enabled. I had convinced myself that all the parameter types where we assign a None default were automatically being tagged Optional by mypy (since clearly we are initializing with None). For that reason I stopped bothering with the explicit Optional. Maybe I was deluding myself.

Reply:
Interesting!  Looks like you actually discovered a MyPy feature I didn't know about: https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-no-implicit-optional

But it sounds like disabling this behavior may become the default in the future, so we might want to be explicit with `Optional` anyway (and add that flag to our config).

Reply:
Ok. I will make it explicit but it seems weird to me since it's so clear that I clearly mean Optional from the defaulting.